### PR TITLE
Fixed #27370 -- Prevented Select widget from using 'required' with a non-empty first value.

### PR DESCRIPTION
**Pull request by timgraham**
https://code.djangoproject.com/ticket/27370

### CODE REVIEW

File: django/forms/widgets.py [hunk 1]
```diff
@@ -680,6 +680,27 @@ def get_context(self, name, value, attrs=None):
...
+        if self.allow_multiple_selected:
+            return use_required_attribute
+
+        def choice_has_empty_value(choice):
+            """Return True if the choice's value is empty string or None."""
+            value, _ = choice
+            return (
+                (isinstance(value, six.string_types) and not bool(value)) or
+                value is None
+            )
```
Comment (charettes) [inline django/forms/widgets.py]: Might be worth defining a non local function for this purpose as it's going to be used on every form rendering?

Comment (timgraham) [inline django/forms/widgets.py]: e.g. `_choice_has_empty_value` static method?

Comment (charettes) [inline django/forms/widgets.py]: yeah, as long as we don't create a function on every single call.

**Comment by timgraham** [comment]
Would it be better to put the tests in `forms_tests/widget_tests` and unit test the affected method rather than to test this via field rendering (considering the field wasn't changed at all)?

**Comment by charettes** [comment]
@timgraham good point, `forms_tests/widget_tests` makes more sense to me as well.

**Pull request merged**